kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/internal/coordinatedcommits/CommitCoordinatorClientSuite.scala

```diff
@@ -0,0 +1,276 @@
+      assert(
+        ex.getMessage.contains(
+          "BackfillToVersion not implemented in TestCommitCoordinatorClient for logPath"))
+    }, Map(CommitCoordinatorProvider.getCommitCoordinatorNameConfKey("cs-name") ->
+      classOf[Builder4].getName))
+  }
+
+  test("set CommitCoordinator config to a class that doesn't extend CommitCoordinator") {
+    val hadoopConf = new Configuration()
+    hadoopConf.set(
+      CommitCoordinatorProvider.getCommitCoordinatorNameConfKey("fake"),
+      "java.lang.String")
+    val e = intercept[IllegalArgumentException](
+      CommitCoordinatorProvider.getCommitCoordinatorClient(
+        hadoopConf, "fake", Collections.emptyMap())
+    )
+    assert(e.getMessage.contains(
+      "Can not instantiate `CommitCoordinatorBuilder` class (from config): %s"
+        .format("java.lang.String")))
+  }
+}
+
+protected trait TestCommitCoordinatorClientBase extends CommitCoordinatorClient {
+  override def registerTable(
+    logPath: Path,
+    currentVersion: Long,
+    currentMetadata: AbstractMetadata,
+    currentProtocol: AbstractProtocol): util.Map[String, String] = {
+    throw new UnsupportedOperationException("Not implemented")
+  }
+
+  override def commit(
+    logStore: LogStore,
+    hadoopConf: Configuration,
+    logPath: Path,
+    coordinatedCommitsTableConf: util.Map[String, String],
+    commitVersion: Long,
+    actions: util.Iterator[String],
+    updatedActions: UpdatedActions): CommitResponse = {
+    throw new UnsupportedOperationException("Not implemented")
+  }
+
+  override def getCommits(
+    logPath: Path,
+    coordinatedCommitsTableConf: util.Map[String, String],
+    startVersion: lang.Long,
+    endVersion: lang.Long = null): GetCommitsResponse =
+    new GetCommitsResponse(Collections.emptyList(), -1)
+
+  override def backfillToVersion(
+    logStore: LogStore,
+    hadoopConf: Configuration,
+    logPath: Path,
+    coordinatedCommitsTableConf: util.Map[String, String],
+    version: Long,
+    lastKnownBackfilledVersion: lang.Long): Unit = {}
+
+  override def semanticEquals(other: CommitCoordinatorClient): lang.Boolean = this == other
+}
+
+// Test 1
+private[coordinatedcommits] class TestCommitCoordinatorClient1
+  extends TestCommitCoordinatorClientBase
+private[coordinatedcommits] class TestCommitCoordinatorClient2
+  extends TestCommitCoordinatorClientBase
+
+object TestCommitCoordinatorClientInstances {
+  val cs1 = new TestCommitCoordinatorClient1()
+  val cs2 = new TestCommitCoordinatorClient2()
+}
+
+class Builder1(hadoopConf: Configuration) extends CommitCoordinatorBuilder(hadoopConf) {
+  override def build(conf: util.Map[String, String]): CommitCoordinatorClient = {
+    conf.getOrElse("url", "") match {
+      case "url1" => TestCommitCoordinatorClientInstances.cs1
+      case "url2" => TestCommitCoordinatorClientInstances.cs2
+      case _ => throw new IllegalArgumentException("Invalid url")
+    }
+  }
+  override def getName: String = "cs-x"
+}
+
+// Test 2
```

Comment:
can you add a bit more details on what's special about this client? Same comment for other test clients.